### PR TITLE
Subdivider

### PR DESCRIPTION
**Pull request by tjlaboss**
Allows the user to subdivide a Universe along a uniform rectangular mesh.

I'e been using this extensively in my own research to perform spatial mesh refinement on scales too find for the CMFD solver, or for when I need to refine specific fuel assemblies more than the whole problem. The advantage of doing this over using azimuthal rings and sectors is that this will work on generic geometries (not just _z_-cylinders).

Results for the unit test were generated by manually dividing a fuel pin and its moderator cell into 4 quadrants (8 FSRs). Subdividing this pin on a 2x2 mesh must be identical to subdividing the whole pin (not adding any fringe FSRs).

flat-source-regions-z-0

Currently this only works for subdividing in the _x_- and _y_- directions. Adding axial division for a 3D lattice will be implemented in a future PR.

This addresses Issue #340 .

### CODE REVIEW

File: openmoc/subdivider.py [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+    
+    def _to_openmoc_universe(self, name=""):
+        if name:
+            mesh_name = "x".join(np.array(self.division, dtype=str))
+            name += " (subdivided " + mesh_name + ")"
+        uout = openmoc.Universe(name=name)
+        cout = openmoc.Cell()
+        cout.setFill(self)
+        uout.addCell(cout)
+        return uout
```
Comment (GiudGiud) [inline openmoc/subdivider.py]: why return a universe containing a cell containing the lattice ? And not just the lattice itself

File: openmoc/subdivider.py [hunk 2]
```diff
@@ -0,0 +1,124 @@
+# Subdividing Lattice
+#
+# Class that performs spatial discretization on a mesh
```
Comment (GiudGiud) [inline openmoc/subdivider.py]: other Python files don't have a header.
If we're going to have one, let's have a clean nice one like for .cpp files

File: openmoc/subdivider.py (hunk 1 above)
Comment (tjlaboss) [reply openmoc/subdivider.py]: It doesn't seem to work unless I put the lattice in another cell.

```
terminate called after throwing an instance of 'std::logic_error'
  what():  Could not find a Cell containing the start Point of this Track:
...  Track: start, x = 1.91667, y = -2, end, x = 2, y = -1.91667, 
... phi = 0.785398
```

As for returning the universe, I want universe in -> universe out for consistency.

**Comment by tjlaboss** [comment]
Thanks for the review, I've resolved 1 and 3 in the latest commit.

After 544555d, the new test will now run in Python 2, but I can't tell why it's failing yet.

**Comment by GiudGiud** [comment]
So when upper_left and lower_right aren't [0.5, 0.5] in your test, the ray tracer gets lost afterwards, presumably because the lattice that contains the universes has some holes now. 
I don't see a simple check to prevent that though.

**Comment by tjlaboss** [comment]
Closing and reopening to run TravisCI w/o the Python2 tests

**Pull request merged**